snapshots/overlay/overlay.go

```diff
@@ -572,3 +605,23 @@ func supportsIndex() bool {
 	}
 	return false
 }
+
+// getMetacopyState returns metacopy state according to kernel support and
+// overlayfs mount options.
+func getMetacopyState(metacopy bool, userxattr bool) MetacopyState {
+	if _, err := os.Stat("/sys/module/overlay/parameters/metacopy"); err != nil {
+		return metacopyUnsupported
+	}
+
+	if !metacopy {
+		logrus.Infof("Metacopy disabled.")
+		return metacopyDisabled
+	}
+	if userxattr {
+		logrus.Warnf("userxattr, and metacopy=on can't be used together. Metacopy disabled.")
```

Comment:
The userxattr option is set in case of we try to create unprivileged mount point of overlayfs, for example in case of containerd is running inside user namespace under user that is not a real root. @AkihiroSuda, please correct me if I am wrong. In this case we will be unable to create overlayfs mount point with metacopy=on.

Regarding error handling I've just following the userxattr option handling approach.
If mounting of rootfs with metacopy=on is impossible, this is not a critical failure. So, we can log warning here.
Dear reviewers @AkihiroSuda @dmcgowan , could you please share your opinion here?

Thanks.

Reply:
Also, maybe it is necessary to add a sanity check for redirect_dir, due to metacopy=on will not work if redirect_dir={off|nofollow|follow(*)}. This is from https://www.kernel.org/doc/Documentation/filesystems/overlayfs.txt.

But currently containerd doesn't touch redirect_dir, due to this fact I drop this check.

Reply:
Warn looks good

Reply:
Thanks @AkihiroSuda!